test/spec/lateralus.js

```diff
@@ -359,6 +359,36 @@ define([
               assert.deepEqual(collectedTest, [5]);
             });
           });
+
+          describe('Returned array contains no undefined values', function () {
+            var App = getLateralusApp();
+
+            _.extend(App.prototype, {
+              provide: {
+                test: function () {
+                  return;
+                }
+              }
+            });
+
+            var ExtendedComponent = Lateralus.Component.extend({
+              name: 'extended'
+              ,provide: {
+                test: function () {
+                  return 1;
+                }
+              }
+            });
```

Comment:
If I were you, which I'm not, I would probably have an additional item in here, just to cover my paranoia regarding "what if it false positively only ever returns one item and isn't actually processing all components" -- if you can assert `[1, 'foo']` it feels much safer. $0.02!